source/agora/network/NetworkManager.d

```diff
@@ -365,6 +381,8 @@ public class NetworkManager
             try
             {
                 node.handshake(this.getAddress());
+                if (node.quorum_hash != StellarHash.init)
+                    this.quorum_sets[node.quorum_hash] = node.quorum_set;
```

Comment:
Actually this should be lazy. If the quorum hash is already known, no need to retrieve the quorum set.